Front end routing realignment

kyle1morel opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
# Description

This PR aligns the front end routing with the newly designed routing tree
- Routing is appropriately nested to allow vue router to build correct matched paths
- Internal and external route paths now begin with `/e` and `/i`
- Route names now contain `EXT_` and `INT_` prefixes to differentiate between public facing and internal government routes
- Navigational permissions are now split into global, external, and internal routes
- Navigational permissions seriously simplified due to new nesting approach
- Routing tree available at: https://miro.com/app/board/uXjVLjp0r78=/
- Disabled the breadcrumb. It's a big WIP
- Removed some unused files

<!-- Describe your changes in detail -->
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->

## Types of changes

<!-- What types of changes does your code introduce? Uncomment all that apply: -->

Bug fix (non-breaking change which fixes an issue)
New feature (non-breaking change which adds functionality) 
Documentation (non-breaking change with enhancements to documentation)
<!-- Breaking change (fix or feature that would cause existing functionality to change) -->

## Checklist

<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->

slhurley commented:
A couple of bugs:
- https://pcns-dev-pr-271.apps.silver.devops.gov.bc.ca/i/housing shows the old "My drafts and previous entries" page when flipping to Proponent on front-end from Navigator role
- Navigator Read Only seems to have write privileges
- Proponent view logged in as BCSC...my draft did not get deleted after I submitted so I have both the Submission and the Draft in my list and if I view the Draft for too long I will get a second draft. 
- If you are in a specific Enquiry, it should return you to the Enquiry tab on the submissions view and not to the Projects tab
- If you create a BF note, the back buttons fire all kinds of continuous "Missing required param "submissionId" (or enquiryID) toasts.  Only BF notes.  If you delete the note altogether it is fine.
- You can't use the "Create Submission" button as a Nav.  It doesn't fill in the contact info and you can't edit the contact info on the submission screen.  I believe we had fixed this before.

qhanson55 commented on frontend/src/App.vue:
```diff
@@ -1,18 +1,19 @@
-import { RouterView, useRouter } from 'vue-router';
+import { RouterView, useRoute } from 'vue-router';
 
+//import Breadcrumb from './components/common/Breadcrumb.vue';
```
remove commented code

qhanson55 commented on frontend/src/views/HomeView.vue:
```diff
@@ -2,11 +2,14 @@
 const toHousing = (): void => {
-  router.push({ name: RouteName.HOUSING });
+  if (useAuthZStore().canNavigate(NavigationPermission.INT_HOUSING)) router.push({ name: RouteName.INT_HOUSING });
+  else router.push({ name: RouteName.EXT_HOUSING });
```
users will have either ext and int navigation permissions but on the off chance they have neither should we be checking that they have ext permissions and not just an else. If neither do nothing or show toast showing say you have no permissions or something

kyle1morel commented:
> * https://pcns-dev-pr-271.apps.silver.devops.gov.bc.ca/i/housing shows the old "My drafts and previous entries" page when flipping to Proponent on front-end from Navigator role
Removed, although the proponent side wouldn't see this

> * Navigator Read Only seems to have write privileges
Non issue. The front end role viewer only affects navigational permissions, not back end permissions.

> * Proponent view logged in as BCSC...my draft did not get deleted after I submitted so I have both the Submission and the Draft in my list and if I view the Draft for too long I will get a second draft.
Unable to replicate.

> * If you are in a specific Enquiry, it should return you to the Enquiry tab on the submissions view and not to the Projects tab
Not something that changed with this PR. State saving between views is a little non standard, please write up a ticket if this is desired functionality.

> * If you create a BF note, the back buttons fire all kinds of continuous "Missing required param "submissionId" (or enquiryID) toasts.  Only BF notes.  If you delete the note altogether it is fine.
Resolved.

> * You can't use the "Create Submission" button as a Nav.  It doesn't fill in the contact info and you can't edit the contact info on the submission screen.  I believe we had fixed this before.
Resolved.

qhanson55 commented:
> * If you are in a specific Enquiry, it should return you to the Enquiry tab on the submissions view and not to the Projects tab
This is what I was talking about yesterday, in regards to PADS-250
